src/main/java/executor/service/service/impl/parallel/ScenarioSourceQueue.java

```diff
@@ -0,0 +1,44 @@
+package executor.service.service.impl.parallel;
+
+import executor.service.model.Scenario;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+/**
+ * The {@code ScenarioSourceQueue} class represents a thread-safe queue for storing
+ * and retrieving instances of {@link Scenario}.
+ * It provides methods for putting and getting scenario configurations.
+ * <p>
+ * This class uses a blocking queue internally to ensure thread safety when
+ * adding and retrieving scenario configurations.
+ * <p>
+ *
+ * @author Oleksandr Tuleninov
```

Comment:
Not sure about correctness of author representation, since [here](https://github.com/GFLCourses5/worker/blob/spring/src/main/java/executor/service/model/ScenarioQueue.java)
and [here](https://github.com/GFLCourses5/worker/blob/beta-main/src/main/java/executor/service/queue/ScenarioQueue.java) same class was proposed earlier.